### PR TITLE
hugo import from jekyll See #101

### PR DESCRIPTION
**Pull request by coderzh**
Issue  #101 
```
hugo import jekyll jekyll_root_path target_path
```
Implemented:
 * Create new hugo site
 * Create config.yaml
 * Convert all markdown contents.
 * Copy all other files and folders to `static`

### CODE REVIEW

**Comment by bep** [comment]
This is pure gold! We should test this with a few Jekyll sites on both *nix and Windows, but this will be a big sell for Hugo.

/cc @spf13

**Comment by bep** [comment]
I tested this on OSX by converting the two first sites on

http://jekyllrb.com/docs/sites/

It worked (almost) flawless. One question: The below site have images under /images. I have no idea what the Jekyll standard says about this, but for this particular case, the images doesn't get copied to /static.

https://github.com/qrush/qrush.github.com

As I said, not sure if this is correct or not.

**Comment by coderzh** [comment]
@bep Thank you for your thorough testing. Yes, I missed copy some files and folders. The Jekyll structure rule as below:
http://jekyllrb.com/docs/structure/

I have added copyJekyllFilesAndFolders function to fix it, please check it again, thx!

**Comment by bep** [comment]
Yes, now it works great.

One comment, though, re. below.

`simple-hugo` is a nice theme, but it is out of our control. It would be better to use a theme in the `spf13` repo as an example, maybe the Hyde theme?

Also, isn't `git submodule` preferred over clones in this case? Not sure. @spf13 ??

```
rm -rf qr2
➜  dump  hugo import jekyll qrush.github.com qr2
importing...
Congratulations! 72 posts imported!
Now, start hugo by yourself:
$ git clone https://github.com/druzza/simple-hugo.git qr2/themes/simple-hugo
```

**Comment by anthonyfok** [comment]
@coderzh, awesome job with the importer!  Just like @bep, this is pure gold!  Thank you for your contribution.

I have a big favour to ask you though.  Since this is a significantly large contribution, please read http://discuss.gohugo.io/t/switching-to-apache-2-license/173 and sign there if you agree.  We would like to switch from "SimPL-2.0" to "Apache License 2.0" in the near future (hopefully).  Many thanks!

**Comment by coderzh** [comment]
@bep The reason why I chose `simple-hugo` is that it is very simple and clean, I tested `Hyde` before, it displays a list of whole content of posts in Index Page. When the number of Posts is very large, it doesn't display very good.

And I have found another perfect theme in `spf13` repo called `herring-cove`. As mentioned in:
https://gohugo.io/tutorials/automated-deployments/

So I changed `simple-hugo` to `herring-cove`,  please have a check, thx!

As the automated-deployments page said, rm .git folder in the theme folder after clone sounds like the official  method?

**Comment by coderzh** [comment]
@anthonyfok Signed already, thx!

**Comment by bep** [comment]
@coderzh yes, my main issue about the theme was its location. Now we should be all good to go (some documentation will fall in place). Any objections, @spf13 ?

**Comment by coderzh** [comment]
Just modified some code about Error Handling.

**Comment by bep** [comment]
Could you also fix the commit message in line with the guidelines?

File: commands/import.go [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+	jww.SetStdoutThreshold(jww.LevelWarn)
+
+	if len(args) < 2 {
+		jww.ERROR.Println(`Import jekyll requires two path, e.g. ` + "`hugo import jekyll jekyll_root_path target_path`.")
+		return
+	}
+
+	jekyllRoot, err := filepath.Abs(filepath.Clean(args[0]))
+	if err != nil {
+		jww.ERROR.Println("Path error:", args[0])
```
Comment (bep) [inline commands/import.go]: Should return here. No point in going in with no Jekyll source.

File: commands/import.go [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+	}
+
+	targetDir, err := filepath.Abs(filepath.Clean(args[1]))
+	if err != nil {
+		jww.ERROR.Println("Path error:", args[1])
```
Comment (bep) [inline commands/import.go]: Return.

File: commands/import.go (hunk 1 above)
Comment (coderzh) [inline commands/import.go]: Done.

File: commands/import.go (hunk 2 above)
Comment (coderzh) [inline commands/import.go]: Done.

**Comment by coderzh** [comment]
All done.

**Comment by spf13** [comment]
I haven't tested it yet, but it looks great and I know users will be
thrilled to have this.

**Comment by bep** [comment]
Merged in 9a2f6c6

Real life testing will probably lead to more requirements, but this is a fantastic start.

**Comment by vielmetti** [comment]
This is super. (And there's an existing path from Typepad to Jekyll, using https://github.com/koozie/typepad_to_jekyll , so this will be helpful for those making that transition).

**Comment by OJ** [comment]
Love it! This is super exciting and is the final piece of the puzzle for getting me off Octopress. Great effort, thanks very much to all involved. :clap:

**Pull request closed without merging**